client/client.go

```diff
@@ -215,6 +216,16 @@ func defaultHTTPClient(hostURL *url.URL) (*http.Client, error) {
 	}, nil
 }
 
+// tlsConfig returns the TLS configuration from the client's transport.
+// It returns nil if the transport is not a [http.Transport], or if no
+// TLSClientConfig is set.
+func (cli *Client) tlsConfig() *tls.Config {
```

Comment:
Small note; I put this function _after_ ` defaultHTTPClient()` so that all methods on the Client are together, and to put it closer to where it's used 😅 (@cpuguy83's PR put it above `defaultHTTPClient()`